### PR TITLE
monitoring/definitions: Make repo updater readable

### PR DESCRIPTION
**Pull request by indradhanush**
I was working on adding a new observable to this file and found it difficult to parse and understand the structure as a first time contributor to this part of the codebase. Refactored it to make it easier to read.

Easier viewed without [hidden whitespace changes](https://github.com/sourcegraph/sourcegraph/pull/24092/files?diff=unified&w=1).

### CODE REVIEW

File: monitoring/definitions/repo_updater.go [hunk 1]
```diff
@@ -8,14 +8,409 @@ import (
...
-		containerName = "repo-updater"
+const (
+	containerName = "repo-updater"
+
+	// This is set a bit longer than maxSyncInterval in internal/repos/syncer.go
+	syncDurationThreshold = 9 * time.Hour
+)
+
+// Collection of ordered monitoring.Observable vars for the "Repositories" monitoring.Group
+var (
```
Comment (bobheadxi) [inline monitoring/definitions/repo_updater.go:19]: All variables declared here are global to all monitoring definitions - can we scope these inside `func RepoUpdater()`?

**Review by artemruts — APPROVED** [review]

**Review by bobheadxi — CHANGES_REQUESTED** [review]
This is a significant departure from how monitoring definitions are set up for all services - we try to keep things explicit and inline for clarity so that when looking for definitions, we don't have to chase down variables etc (where reasonable - notable exceptions being shared configuration)

If this is difficult to read, I would advise updating `Name` and `Descriptions` to be easier to search

**Comment by bobheadxi** [comment]
On a friendlier note 😄  I'm open to changes in best practices for defining monitoring, but I'm finding it a bit difficult to see how this is more readable - there is still a lot of struct definitions, except now it's even harder to see where each definition lives (which group? which row?), and the variable names do not seem more informative to the uninformed (me!) because now I have to go to the definition to see what's going on.

**Comment by bobheadxi** [comment]
> I was working on adding a new observable to this file and found it difficult to parse and understand the structure as a first time contributor to this part of the codebase

I'd love to improve on this as well! We have a lot of [inline documentation](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/docs/monitoring/monitoring#Container) outlining how things are laid out internally, as well as a [guide](https://docs.sourcegraph.com/dev/how-to/add_monitoring#alerts-dashboards-and-documentation) for how to add monitoring - let me know if there's anything that can be better on this front!

**Comment by indradhanush** [comment]
Thank you for the feedback. When I approached adding a new observable in this file, it was difficult for me to figure out where each `monitoring.Row` and `monitoring.Group` started and finished. I felt it might be easier to create variables so that at a glance it was easier to look at the collection of `Observables` in each row and group respectively and then lookup definition based on whichever metric the reader is concerned about. 

For now, I will go ahead and close this PR and as I work more in this part of the codebase I will collect a list of pain points if I happen to run across and open it for discussion so that we can decide on a cleaner approach and continue to keep the structure of our monitoring definitions consistent. In hindsight, it doesn't make sense to change just a single area without finalising on a more consistent style and approach.

**Pull request closed without merging**